server/webapp/WEB-INF/rails/webpack/views/pages/config_repos/spec/config_repo_view_model_spec.ts

```diff
@@ -0,0 +1,82 @@
+/*
+* Copyright 2019 ThoughtWorks, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import SparkRoutes from "helpers/spark_routes";
+import * as stream from "mithril/stream";
+import {Stream} from "mithril/stream";
+import {ConfigRepo} from "models/config_repos/types";
+import {PluginInfo} from "models/shared/plugin_infos_new/plugin_info";
+import {FlashContainer, RequiresPluginInfos, SaveOperation} from "views/pages/page_operations";
+import {ConfigRepoVM} from "../config_repo_view_model";
+import {mockResultsCache} from "./test_data";
+
+describe("ConfigRepoVM", () => {
+  it("fetches data on the expand event", () => {
+    const vm = createVm();
+    expect(vm.results.prime).not.toHaveBeenCalled();
+
+    vm.notify("expand");
+
+    expect(vm.results.prime).toHaveBeenCalled();
+  });
+
+  it("invalidates and updates data on refresh event", () => {
+    const vm = createVm();
+
+    expect(vm.results.invalidate).not.toHaveBeenCalled();
+
+    vm.notify("refresh");
+
+    expect(vm.results.invalidate).toHaveBeenCalled();
+  });
+
+  it("reparseRepo() triggers reparse update invalidates the results cache", (done) => {
```

Comment:
reword?

Reply:
I was obviously ~drunk~ too tired when I wrote this.